projects/fal/src/fal/file_sync.py

```diff
@@ -314,10 +314,12 @@ def sync_files(
 
         # Categorize based on server response
         files_to_upload: List[FileMetadata] = []
+        hashes_to_upload = set()
```

Comment:
would it make more sense to change files_to_upload to be a dict of hash -> metadata